coreneuron/io/mech_report.cpp

```diff
@@ -19,7 +19,7 @@ namespace coreneuron {
 void write_mech_report() {
     /// mechanim count across all gids, local to rank
     const auto n_memb_func = corenrn.get_memb_funcs().size();
-    std::vector<unsigned long long> local_mech_count(n_memb_func, 0);
+    std::vector<size_t> local_mech_count(n_memb_func, 0);
```

Comment:
It seems a bit messy to use `size_t` here and then make `nrnmpi_long_allreduce_vec` expect `unsigned long` when `nrnmpi_long_allreduce` expects `long`.

Might it be easier/more consistent to just use `long` and sidestep issues like whether `size_t*` -> `unsigned long*` is OK, and whether `MPI_UNSIGNED_LONG_LONG` is the correct value for `size_t`?

Reply:
I did somewhat sloppy job here. I started changing stuff and quickly I realised that types are used inconsistently at multiple places. So I end-up creating #491.

> Might it be easier/more consistent to just use long and sidestep issues like whether

Considering these routines are mostly used for statistics printing and `long` should be sufficiently large (on 64bit OS), I will replace size_t with long as you suggested.